dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java

```diff
@@ -180,6 +184,53 @@ public void testAboutUsDefaultPageToLanguageFalse() throws Exception { // https:
         CacheLocator.getNavToolCache().removeNav(demoHost.getIdentifier(), aboutUsFolder.getInode(), 2);
     }
 
+    @Test
+    public void testRootLevelNavigation_WhenOneFileAssetIsShownOnMenu() throws Exception {
+
+        //Using System User.
+        final User user = APILocator.getUserAPI().getSystemUser();
+
+        //Get SystemFolder
+        final Folder systemFolder = APILocator.getFolderAPI().findSystemFolder();
+
+        //Using demo.dotcms.com host.
+        final Host demoHost = APILocator.getHostAPI().findByName("demo.dotcms.com", user, false);
+
+        //Create a FileAsset In English With ShowOnMenu = true
+        final File file = File.createTempFile("fileTestEngTrue", ".txt");
+        FileUtil.write(file, "helloworld");
+        final FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(systemFolder, file);
+        final Contentlet fileAsset = fileAssetDataGen.nextPersisted();
+        Contentlet fileAssetShown = APILocator.getContentletAPI().find(fileAsset.getInode(),user,false);
+        fileAssetShown.setBoolProperty(FileAssetAPI.SHOW_ON_MENU,true);
+        fileAssetShown.setInode("");
+        fileAssetShown = APILocator.getContentletAPI().checkin(fileAssetShown,user,false);
+        APILocator.getContentletAPI().publish(fileAssetShown,user,false);
+        APILocator.getContentletAPI().isInodeIndexed(fileAssetShown.getInode(),true);
+
+        //Create a FileAsset In English With ShowOnMenu = false
+        final File file2 = File.createTempFile("fileTestEngFalse", ".txt");
+        FileUtil.write(file2, "helloworld");
+        final FileAssetDataGen fileAssetDataGen2 = new FileAssetDataGen(systemFolder, file2);
+        final Contentlet fileAssetNotShown = fileAssetDataGen2.nextPersisted();
+        APILocator.getContentletAPI().publish(fileAssetNotShown,user,false);
+        APILocator.getContentletAPI().isInodeIndexed(fileAssetNotShown.getInode(),true);
+
+        //Get the Nav at the Root Level
+        final NavResult navResult = new NavTool().getNav(demoHost,systemFolder.getPath(),1,user);
+        assertNotNull(navResult);
+        assertEquals(1,navResult.getChildren().size());
+
+        //Now remove all the pages that we created for this tests.
```

Comment:
This cleanup should be in a finally statement

Reply:
done